### PR TITLE
[api-extractor] Improve "export * from" to support cyclic references

### PR DESCRIPTION
**Pull request by pgonzal**
Fixes issue https://github.com/Microsoft/web-build-tools/issues/1033

### CODE REVIEW

**Comment by pgonzal** [comment]
@natalieethell with this fix, **office-ui-fabric-react** finally extracts successfully!! 🤤🤤🤤

File: build-tests/api-extractor-scenarios/src/circularImport2/IFolder.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+export * from "./IFile";
```
Comment (iclanton) [inline build-tests/api-extractor-scenarios/src/circularImport2/IFolder.ts]: >" [](start = 14, length = 1)

Double quotes?

File: apps/api-extractor/src/analyzer/AstModule.ts [hunk 2]
```diff
@@ -5,29 +5,43 @@ import * as ts from 'typescript';
...
    */
-  public externalModulePath: string | undefined;
+  public readonly externalModulePath: string | undefined;
+
+  public readonly starExportedModules: Set<AstModule>;
+
+  public readonly cachedExportedSymbols: Map<string, AstSymbol>;
 
-  public constructor(sourceFile: ts.SourceFile) {
+  public astModuleExportInfo: AstModuleExportInfo | undefined;
```
Comment (iclanton) [inline apps/api-extractor/src/analyzer/AstModule.ts]: Document.

**Comment by pgonzal** [comment]
@natalieethell 

This PR was published as the beta release **api-extractor@7.0.11** and **api-documenter@7.0.16**.

To try it, do `npm install @microsoft/api-extractor@beta`

**Pull request merged**